### ISSUE TITLE
Modelator fails with latest Apalache

### ISSUE THREAD

**Issue by rnbguy** [informalsystems/modelator#81]
Recent Apalache versions report counterexamples of model invariants as errors.

```
# Tool home: system
# Package:   
# JVM args:  -Xmx4096m -DTLA-Library=
#
# APALACHE version 0.15.13 build ad76fbf
#
# WARNING: This tool is in the experimental stage.
#          Please report bugs at: [https://github.com/informalsystems/apalache/issues]
# 
# Usage statistics is OFF. We care about your privacy.
# If you want to help our project, consider enabling statistics with config --enable-stats=true.

Checker options: filename=Simple_Test.tla, init=, next=, inv=     I@18:04:41.045
Tuning:                                                           I@18:04:41.242
PASS #0: SanyParser                                               I@18:04:41.243
Parsing file /home/playground/Simple_Test.tla
Parsing file /home/playground/Simple.tla
Parsing file /home/playground/Ops.tla
Parsing file /tmp/Integers.tla
Parsing file /tmp/Naturals.tla
PASS #1: TypeCheckerSnowcat                                       I@18:04:41.520
 > Running Snowcat .::.                                           I@18:04:41.521
 > Your types are great!                                          I@18:04:41.592
 > All expressions are typed                                      I@18:04:41.593
PASS #2: ConfigurationPass                                        I@18:04:41.619
  > Simple_Test.cfg: Loading TLC configuration                    I@18:04:41.622
  > Using the init predicate Init from the TLC config             I@18:04:41.654
  > Using the next predicate Next from the TLC config             I@18:04:41.654
  > Simple_Test.cfg: found INVARIANTS: TestNeg                    I@18:04:41.654
  > Set the initialization predicate to Init                      I@18:04:41.655
  > Set the transition predicate to Next                          I@18:04:41.655
  > Set an invariant to TestNeg                                   I@18:04:41.655
PASS #3: DesugarerPass                                            I@18:04:41.666
  > Desugaring...                                                 I@18:04:41.666
PASS #4: UnrollPass                                               I@18:04:41.677
  > Unroller                                                      I@18:04:41.686
PASS #5: InlinePass                                               I@18:04:41.702
  > InlinerOfUserOper                                             I@18:04:41.703
  > Wrap                                                          I@18:04:41.705
  > CallByNameOperatorEmbedder                                    I@18:04:41.706
  > LetInExpander                                                 I@18:04:41.707
  > Unwrap                                                        I@18:04:41.707
  > InlinerOfUserOper                                             I@18:04:41.707
Leaving only relevant operators: CInitPrimed, Init, InitPrimed, Next, TestNeg I@18:04:41.709
PASS #6: PrimingPass                                              I@18:04:41.714
  > Introducing InitPrimed for Init'                              I@18:04:41.715
PASS #7: VCGen                                                    I@18:04:41.723
  > Producing verification conditions from the invariant TestNeg  I@18:04:41.723
  > VCGen produced 1 verification condition(s)                    I@18:04:41.728
PASS #8: PreprocessingPass                                        I@18:04:41.735
  > Before preprocessing: unique renaming                         I@18:04:41.736
 > Applying standard transformations:                             I@18:04:41.741
  > PrimePropagation                                              I@18:04:41.741
  > Desugarer                                                     I@18:04:41.750
  > UniqueRenamer                                                 I@18:04:41.761
  > Normalizer                                                    I@18:04:41.769
  > Keramelizer                                                   I@18:04:41.780
  > After preprocessing: UniqueRenamer                            I@18:04:41.792
PASS #9: TransitionFinderPass                                     I@18:04:41.799
  > Found 1 initializing transitions                              I@18:04:41.805
  > Found 1 transitions                                           I@18:04:41.808
  > No constant initializer                                       I@18:04:41.808
  > Applying unique renaming                                      I@18:04:41.809
PASS #10: OptimizationPass                                        I@18:04:41.819
 > Applying optimizations:                                        I@18:04:41.822
  > ConstSimplifier                                               I@18:04:41.823
  > ExprOptimizer                                                 I@18:04:41.824
  > ConstSimplifier                                               I@18:04:41.825
PASS #11: AnalysisPass                                            I@18:04:41.828
 > Marking skolemizable existentials and sets to be expanded...   I@18:04:41.829
  > SkolemizationMarker                                           I@18:04:41.829
  > ExpansionMarker                                               I@18:04:41.830
 > Running analyzers...                                           I@18:04:41.831
  > Introduced expression grades                                  I@18:04:41.838
  > Introduced 0 formula hints                                    I@18:04:41.838
PASS #12: PostTypeCheckerSnowcat                                  I@18:04:41.838
 > Running Snowcat .::.                                           I@18:04:41.838
 > Your types are great!                                          I@18:04:41.855
 > All expressions are typed                                      I@18:04:41.856
PASS #13: BoundedChecker                                          I@18:04:41.865
State 0: Checking 1 state invariants                              I@18:04:42.079
Step 0: picking a transition out of 1 transition(s)               I@18:04:42.082
State 1: Checking 1 state invariants                              I@18:04:42.089
Step 1: picking a transition out of 1 transition(s)               I@18:04:42.089
State 2: Checking 1 state invariants                              I@18:04:42.092
Step 2: picking a transition out of 1 transition(s)               I@18:04:42.092
State 3: Checking 1 state invariants                              I@18:04:42.094
Step 3: picking a transition out of 1 transition(s)               I@18:04:42.095
State 4: Checking 1 state invariants                              I@18:04:42.097
Step 4: picking a transition out of 1 transition(s)               I@18:04:42.098
State 5: Checking 1 state invariants                              I@18:04:42.102
Step 5: picking a transition out of 1 transition(s)               I@18:04:42.104
State 6: Checking 1 state invariants                              I@18:04:42.107
State 6: state invariant 0 violated. Check the counterexample in: counterexample1.tla, MC1.out, counterexample1.json E@18:04:42.132
Found 1 error(s)                                                  I@18:04:42.136
The outcome is: Error                                             I@18:04:42.138
Checker has found an error                                        I@18:04:42.139
It took me 0 days  0 hours  0 min  1 sec                          I@18:04:42.139
Total time: 1.142 sec                                             I@18:04:42.139
EXITCODE: ERROR (12)
```

which makes `modelator` report it as a failure too.

https://github.com/informalsystems/modelator/blob/49a905d0356609127b0745bddbace044f3c084dc/modelator/src/model/checker/apalache/mod.rs#L162-L173

We should handle this `EXITCODE` as a special case.

**Comment by andrey-kuprianov** [comment]
we will leave this out of 0.3 release; it's for 0.3.1

**Comment by rnbguy** [comment]
yes. this is for @danwt's eyes. He said he will look at it for the next release.

**Comment by danwt** [comment]
Closed by #96
